File: Julia/vae-kl-divergence.jl
function kl_divergence(mu, log_var)
    var = exp.(log_var)
    kl_element = 1 .+ log_var .- mu .^ 2 .- var
    batch_kl = -0.5 .* sum(kl_element, dims=2)
    return Float64(mean(batch_kl))
end

File: R/vae-kl-divergence.R
kl_divergence <- function(mu, log_var) {
  var <- exp(log_var)
  kl_element <- 1 + log_var - (mu ^ 2) - var
  batch_kl <- -0.5 * rowSums(kl_element)
  mean(batch_kl)
}
